modules/renter/contractor/journal.go

```diff
@@ -0,0 +1,332 @@
+package contractor
+
+// The contractor achieves efficient persistence using a JSON transaction
+// journal. It enables efficient ACID transactions on JSON objects.
+//
+// The journal represents a single JSON object, containing all of the
+// contractor's persisted data. The object is serialized as an "initial
+// object" followed by a series of update sets, one per line. Each update
```

Comment:
ah. One nice property of having each update on its own line is that in the event of corruption, you can seek directly to the next line. I don't do this currently, but I probably should. Otherwise if you fail to read one update, you may fail to read good updates on subsequent lines.